src/rust/src/decoder/mod.rs

```diff
@@ -19,47 +22,107 @@ const CCX_DTVCC_SCREENGRID_ROWS: u8 = 75;
 const CCX_DTVCC_SCREENGRID_COLUMNS: u8 = 210;
 const CCX_DTVCC_MAX_ROWS: u8 = 15;
 const CCX_DTVCC_MAX_COLUMNS: u8 = 32 * 2;
+const CCX_DTVCC_MAX_SERVICES: usize = 63;
+// const CCX_DTVCC_MAX_WINDOWS: usize = 8;
 
 /// Context required for processing 708 data
-pub struct Dtvcc<'a> {
+pub struct Dtvcc {
     pub is_active: bool,
     pub active_services_count: u8,
-    pub services_active: Vec<i32>,
+    pub services_active: [i32; CCX_DTVCC_MAX_SERVICES],
     pub report_enabled: bool,
-    pub report: &'a mut ccx_decoder_dtvcc_report,
-    pub decoders: Vec<&'a mut dtvcc_service_decoder>,
-    pub packet: Vec<u8>,
+    pub report: *mut ccx_decoder_dtvcc_report,
+    pub decoders: [Option<Box<dtvcc_service_decoder>>; CCX_DTVCC_MAX_SERVICES],
```

Comment:
There is no longer a need to use bindings for any 708 related structs/enums, we can define them now in rust itself. We still need to use bindings for `encoder`, `timing` and `report`. But all others should be defined in rust.
This is going to be somewhat bigger and comprehensive change

Reply:
Should that be a separate PR since it seems to be independent of this one?